datumaro/plugins/open_images_format.py

```diff
@@ -92,23 +94,32 @@ def set_parents_from_node(node, category):
         set_parents_from_node(root_node, root_category)
 
     def _load_items(self):
+        image_paths_by_id = {
+            osp.splitext(osp.basename(path))[0]: path
+            for path in find_images(
+                osp.join(self._dataset_dir, 'images'),
+                recursive=True, max_depth=1)
+        }
+
         items_by_id = {}
 
         def load_from(path):
             with self._open_csv_annotation(path) as image_reader:
                 for image_description in image_reader:
                     image_id = image_description['ImageID']
-                    assert _RE_IMAGE_ID.fullmatch(image_id)
-
                     if image_id in items_by_id:
                         raise RepeatedItemError(item_id=image_id)
 
                     subset = image_description['Subset']
                     assert subset in {'test', 'train', 'validation'}
 
+                    if image_id not in image_paths_by_id:
+                        raise ItemWithMissingImageError(
```

Comment:
In Datumaro we don't consider this as an error necessarily. We work with annotations in the first place. An input dataset might not have images at all, but it needs to have their metainfo in annotations.

Reply:
OK; I removed the error and instead made the image optional.